gslib/command.py

```diff
@@ -725,27 +729,47 @@ def RaiseInvalidArgumentException(self):
                (self.command_spec.usage_synopsis, self.command_name))
     raise CommandException(message)
 
-  def ParseSubOpts(self, check_args=False):
+  def ParseSubOpts(self,
+                   check_args=False,
+                   args=None,
+                   update_sub_opts_and_args=True):
     """Parses sub-opt args.
 
     Args:
       check_args: True to have CheckArguments() called after parsing.
-
-    Populates:
-      (self.sub_opts, self.args) from parsing.
+      args: List of args. If None, self.args will be used.
+      update_sub_opts_and_args: True if self.sub_opts and self.args should
```

Comment:
I would rename to `should_update_sub_opts_and_args`

Reply:
Done.